src/mumble/PulseAudio.cpp

```diff
@@ -456,25 +534,25 @@ void PulseAudioSystem::sink_info_callback(pa_context *, const pa_sink_info *i, i
 
 void PulseAudioSystem::write_stream_callback(pa_stream *s, void *userdata) {
 	PulseAudioSystem *pas = reinterpret_cast<PulseAudioSystem *>(userdata);
-	switch (pa_stream_get_state(s)) {
+	switch (pas->pa_stream_get_state(s)) {
 		case PA_STREAM_FAILED:
-			qWarning("PulseAudio: Stream error: %s", pa_strerror(pa_context_errno(pa_stream_get_context(s))));
+			qWarning("PulseAudio: Stream error: %s", pas->pa_strerror(pas->pa_context_errno(pas->pa_stream_get_context(s))));
 			break;
 		default:
 			break;
 	}
 	const pa_buffer_attr *bufferAttr;
-	if ( (bufferAttr = pa_stream_get_buffer_attr(s)) ) {
+	if ((bufferAttr = pas->pa_stream_get_buffer_attr(s))) {
 		g.ao->setBufferSize(bufferAttr->maxlength);
 	}
 	pas->wakeup();
 }
 
 void PulseAudioSystem::read_stream_callback(pa_stream *s, void *userdata) {
 	PulseAudioSystem *pas = reinterpret_cast<PulseAudioSystem *>(userdata);
```

Comment:
I think while refactoring this, we should also give this a more descriptive name. Maybe something like `pulseAudio` or  `pulseSystem`?

Reply:
I would do that in dedicated pull request, because there's quite a lot that can be refactored.